spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/context/properties/ConfigurationPropertiesReportEndpoint.java

```diff
@@ -312,6 +311,13 @@ public void serializeAsField(Object pojo, JsonGenerator jgen, SerializerProvider
 		}
 
 		private boolean isReadable(BeanDescription beanDesc, BeanPropertyWriter writer) {
+			// if the class has the @ConstructorBinding annotation or
+			// one constructor of this class has the @ConstructorBinding annotation,
+			// we can ignore the setter method.
```

Comment:
Unfortunately, that's not what the code should be doing. If constructor binding is to be used, only the parameters of the matching constructor should be used to determine the properties. Accessors (getters) shouldn't be checked at all.